RevenueCatUI/Templates/Components/Root/RootView.swift

```diff
@@ -31,13 +33,61 @@ struct RootView: View {
         ScrollView {
             StackComponentView(viewModel: viewModel.stackViewModel, onDismiss: onDismiss)
         }.applyIfLet(viewModel.stickyFooterViewModel) { stackView, stickyFooterViewModel in
-            stackView.safeAreaInset(edge: .bottom) {
-                StackComponentView(viewModel: stickyFooterViewModel.stackViewModel, onDismiss: onDismiss)
-            }
+            stackView
+                .safeAreaInset(edge: .bottom) {
+                    StackComponentView(
+                        viewModel: stickyFooterViewModel.stackViewModel,
+                        onDismiss: onDismiss,
+                        additionalPadding: EdgeInsets(
+                            top: 0,
+                            leading: 0,
+                            bottom: additionalFooterPaddingBottom,
+                            trailing: 0
+                        )
+                    )
+                }
+                // First we ensure our footer draws in the bottom safe area. Then we add additional padding, so its
+                // background shows in that same bottom safe area.
+                .ignoresSafeArea(edges: .bottom)
+                .onBottomSafeAreaPaddingChange { bottomPadding in
+                    self.additionalFooterPaddingBottom = bottomPadding
+                }
```

Comment:
I thought about "filling up" the bottom safe area with the same background color (`stackViewModel.backgroundColor`) as set on the sticky footer using [`background(_:ignoressafeareaedges:)`](https://developer.apple.com/documentation/swiftui/view/background(_:ignoressafeareaedges:)), but  that won't work if we ever support background gradients or background images. So I went with the additional padding approach instead.